### PR TITLE
Responses to Assessment

### PR DESCRIPTION
**Pull request by baz1389**

### CODE REVIEW

File: lib/assessment.rb [hunk 1]
```diff
@@ -13,28 +13,29 @@ def longest(enum)
   # enumerable supplied (i.e. apply the `+` operator)
   # Assume each entry implements a `+` method
   def sum(enum)
-    # Your code goes here
+    enum..inject{|sum,x| sum + x }
+    # reduce(&:+) i think this works too??
```
Comment (GA-MEB) [inline lib/assessment.rb:17]: Correct! (minus the extra dot)
`Inject` is actually an alias to `reduce` - they're the same.

File: lib/assessment.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ class Assessment
   # enumerable supplied.  Assume each entry implements a length
   # method that returns a Fixnum.
   def longest(enum)
-    # Your code goes here
+    enum.max {|a, b| a.length <=> b.length}
```
Comment (GA-MEB) [inline lib/assessment.rb:8]: Correct.

File: lib/assessment.rb [hunk 3]
```diff
@@ -13,28 +13,29 @@ def longest(enum)
...
+    enum..inject{|sum,x| sum + x }
+    # reduce(&:+) i think this works too??
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the numbers enumerable are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.any?(&:odd?)
```
Comment (GA-MEB) [inline lib/assessment.rb:24]: Correct.

File: lib/assessment.rb [hunk 4]
```diff
@@ -13,28 +13,29 @@ def longest(enum)
...
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # numbers enumerable are even
   def every_even(numbers)
-    # Your code goes here
+    numbers.all?(&:even?)
```
Comment (GA-MEB) [inline lib/assessment.rb:31]: Correct.

File: lib/assessment.rb [hunk 5]
```diff
@@ -13,28 +13,29 @@ def longest(enum)
...
   end
 
   # Question 5
   # Write ruby code that returns an array with all of the
   # entries from the words enumerable capitalized.
   def transform(words)
-    # Your code goes here
+    words.map(&:capitalize)
```
Comment (GA-MEB) [inline lib/assessment.rb:38]: Correct.

File: lib/assessment.rb [hunk 6]
```diff
@@ -43,6 +44,11 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    File.open(ARGV[0]) do |file|
```
Comment (GA-MEB) [inline lib/assessment.rb:47]: You're getting the filename! No need to check for console arguments.

Comment (MattB-GA-Boston) [inline lib/assessment.rb]: Also, you could have chained this: 
`File.open(filename).each_with_object`

File: lib/assessment.rb [hunk 7]
```diff
@@ -43,6 +44,11 @@ def transform(words)
...
+      lwc = file.each_with_object(Hash.new(0)) do |line, memo|
```
Comment (GA-MEB) [inline lib/assessment.rb:48]: Also, you could have chained this: 
`File.open(filename).each_with_object(Hash.new(0))`

File: lib/assessment.rb [hunk 8]
```diff
@@ -43,6 +44,11 @@ def transform(words)
...
+      memo[:lines] += 1
+      end
+    end
   end
-end
+#I know the keyword chomp removes linebreaks
+#and needs to be implemented somewhere ( maybe with >>)
```
Comment (GA-MEB) [inline lib/assessment.rb:54]: Yeah the functionality's missing. Here's one possible solution, from Antony.

``` ruby
File.open(filename).each_with_object('') do |line, memo|
   memo << line.chomp + ' '
end.strip
```

**Comment by GA-MEB** [comment]
5.25 of 6 (-0.75 on Q6)

Nice work, Ben.

**Pull request closed without merging**